state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1776,7 +1872,57 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		let deposit_channel_details = DepositChannelLookup::<T, I>::get(&deposit_address)
 			.ok_or(Error::<T, I>::InvalidDepositAddress)?;
 
+		ensure!(
+			deposit_channel_details.deposit_channel.asset == asset,
+			Error::<T, I>::AssetMismatch
+		);
+
 		let channel_id = deposit_channel_details.deposit_channel.channel_id;
+		let channel_owner = deposit_channel_details.clone().owner;
```

Comment:
We don't need to clone the full details struct just to get the owner.

```suggestion
		let channel_owner = deposit_channel_details.owner.clone();
```